### ISSUE TITLE
Support Next.js

### ISSUE THREAD

**Issue by QuiiBz** [lagonapp/serverless#47]

**Comment by QuiiBz** [comment]
Update: both the `pages` directory and the `app` directory are working!

https://twitter.com/tomlienard/status/1613229200789438464

**Comment by QuiiBz** [comment]
Update: `AsyncLocalStorage` has been merged for a few weeks, but Next.js now expects the runtime to support imports  (and specifically an implementation of `Buffer` from `node:buffer`), which we don’t have at all right now and would make the runtime even more complicated.

https://twitter.com/ascorbic/status/1648264283824742401?s=46&t=vptTziexWvl21M22w1Jv0w

**Comment by pi0** [comment]
Any reason not leveraging [unjs/unenv](https://github.com/unjs/unenv) or nitro to bundle output?

**Comment by QuiiBz** [comment]
Great question! Implementing these APIs (`AsyncLocalStorage`, `Buffer`...) should be done in the runtime itself because it needs to be very performant. They will ultimately be implemented natively (instead of in JS-land right now) so it doesn't really make sense to use unenv now and remove it later.

**Comment by pi0** [comment]
Yes makes sense for ALS cannot be implemented in Userland. (unenv will also auto-detect such environments and use native soon. It only has a shim. But `Buffer` and other Node.js APIs used by next.js or user dependencies can possibly use can be leveraged unenv userland impl same as Nuxt.
